autoreject/autoreject.py

```diff
@@ -1218,26 +1218,55 @@ def __init__(self, bad_epochs, labels, ch_names):
         assert len(bad_epochs) == labels.shape[0]
         assert len(ch_names) == labels.shape[1]
 
-    def plot(self):
-        """Plot."""
+    def plot(self, orientation='vertical', show=True):
+        """Plot.
+
+        Parameters
+        ----------
+        orientation : 'vertical' or 'horizontal'
+            If `'vertical'`, will plot sensors on x-axis and epochs on y-axis.
+            If `'horizontal'`, will plot epochs on x-axis and sensors
+            on y-axis.
+        show : bool
+            Display the figure immediately.
+
+        Returns
+        -------
+        figure : Instance of matplotlib.figure.Figure
+        """
         import matplotlib.pyplot as plt
 
-        plt.figure(figsize=(12, 6))
-        plt.imshow(self.labels, cmap='Reds',
-                   interpolation='nearest')
-        # XXX to be fixed
-        ch_names_ = self.ch_names[7::10]
+        figure = plt.figure(figsize=(12, 6))
         ax = plt.gca()
         ax.grid(False)
-        ax.set_xlabel('Channels')
-        ax.set_ylabel('Epochs')
-        plt.setp(ax, xticks=range(7, self.labels.shape[1], 10),
-                 xticklabels=ch_names_)
+        ch_names_ = self.ch_names[7::10]
+
+        if orientation == 'horizontal':
+            plt.imshow(self.labels.T, cmap='Reds',
+                       interpolation='nearest')
+            ax.set_xlabel('Epochs')
+            ax.set_ylabel('Channels')
+            plt.setp(ax, yticks=range(7, self.labels.shape[1], 10),
+                     yticklabels=ch_names_)
+        elif orientation == 'vertical':
+            plt.imshow(self.labels, cmap='Reds',
+                       interpolation='nearest')
+            ax.set_xlabel('Channels')
+            ax.set_ylabel('Epochs')
+            plt.setp(ax, xticks=range(7, self.labels.shape[1], 10),
+                     xticklabels=ch_names_)
+        else:
+            msg = "Orientation can be only 'horizontal' or 'vertical'."
```

Comment:
```suggestion
            msg = "orientation can be only 'horizontal' or 'vertical'. Got %s" % orientation
```

Sorry, one last fix request!